### PR TITLE
Fix enterprise mobile form submissions report

### PR DESCRIPTION
**Pull request by millerdev**
https://dimagi.atlassian.net/browse/SAAS-15300

### Safety story

This is a tiny change to make a report more resilient to various form formats.

### Automated test coverage

No.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by gherceg — APPROVED** [review]
👍🏻

File: corehq/apps/enterprise/enterprise.py [hunk 1]
```diff
@@ -242,10 +242,12 @@ def rows_for_domain(self, domain_obj):
         apps = {a.id: a.name for a in apps}
         rows = []
         for hit in self.hits(domain_obj.name):
+            if hit['form'].get('#type') == 'system':
```
Comment (gherceg) [inline corehq/apps/enterprise/enterprise.py:245]:
```suggestion
            if "#type" in hit['form'].keys() and hit['form'].get('#type') == 'system':
```
Would something like this make sense? Or if type isn't on the form would we want it to fail hard?

Comment (millerdev) [reply corehq/apps/enterprise/enterprise.py:245]: `.get()` will return `None` if it's not there.

Comment (gherceg) [reply corehq/apps/enterprise/enterprise.py:245]: Ah of course

**Comment by kishansampat** [comment]
Thanks @millerdev and @gherceg ! I wanted to confirm, does this change just ensure that the report works with the system forms, or is it also taking into account what Kai and Amy mentioned here: https://dimagi.atlassian.net/browse/SAAS-15300?focusedCommentId=311791 that we should exclude system forms from this report? https://dimagi.atlassian.net/browse/SAAS-15300?focusedCommentId=312230

Let me know if the latter needs to be a separate ticket / effort. Graham, I see that you mentioned that you're not aware of any way to exclude system forms.

One more question I had was that submission of system forms is quite common I believe (data cleaning, imports, archival of forms and automatic updates). How come we're facing this error only now and have not seen this in the past?

File: corehq/apps/enterprise/enterprise.py [hunk 2]
```diff
@@ -242,10 +242,12 @@ def rows_for_domain(self, domain_obj):
...
+                continue
             username = hit['form']['meta']['username']
             submitted = self.format_date(datetime.strptime(hit['received_on'][:19], '%Y-%m-%dT%H:%M:%S'))
             rows.append([
-                hit['form']['@name'],
+                hit['form'].get('@name', _('Unnamed')),
```
Comment (jingcheng16) [inline corehq/apps/enterprise/enterprise.py:250]: Just curious, since we will exclude system form, is this line still necessary?

**Comment by millerdev** [comment]
@kishansampat it's taking into account what Kai and Amy mentioned, system forms are excluded from the report.

> How come we're facing this error only now and have not seen this in the past?

Good question. I don't know. It seems possible that (although I don't know if) some system forms have a "name", which would mean the report would not have crashed on those.

**Pull request merged**